ui/components/app/edit-gas-fee-popover/network-status/status-slider/status-slider.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+const GradientColors = [
+  '#037DD6',
+  '#1876C8',
+  '#2D70BA',
+  '#4369AB',
+  '#57629E',
+  '#6A5D92',
+  '#805683',
+  '#9A4D71',
+  '#B44561',
+  '#C54055',
+];
+
+const StatusSlider = () => {
+  const statusValue = 0.5;
+  const sliderValueNumeric = Math.round(statusValue * 10);
+  return (
+    <div className="status-slider">
+      <div
+        className="status-slider__arrow"
+        style={{
+          borderTopColor: GradientColors[sliderValueNumeric],
+          marginLeft: `${sliderValueNumeric * 10 - 10}%`,
```

Comment:
It looks like `statusValue` is already being multiplied by 10 — did you mean to divide `sliderValueNumeric` by 10 again or not multiply this by 10 here?

Also, what is the reasoning behind subtracting 10 here? If it's because the X coordinate of the arrow starts at the left side of the arrow, maybe it's worth adding a negative margin-left to the arrow? This way the positioning of the arrow is anchored at its center and you may not need an extra -10% here.

Reply:
This is to account for the width of arrow.

Reply:
Gotcha, that's what I figured. Since this is disconnected from the CSS, that might not be obvious. What if you were to add a negative margin-left to the arrow element so that when `marginLeft` here is 0% then the point of the arrow lands at the right spot?

Reply:
Total value of margin-left depends on `statusValue` so for status value .5 it will be 40%.

Not sure how do you suggest doing it.

Reply:
Okay, I played around with this a bit and came up with an idea where the arrow is wrapped in a couple of divs. The reason for `arrow-border` is that I noticed in the designs that there is a small white border around the arrow, and it gets pressed into the slider a bit. The reason for `arrow-wrapper` is that it allows you to apply a margin-left in a clear way without having to do any subtraction. Let me know what you think about this:

``` jsx
<div
  className="status-slider__arrow-wrapper"
  style={{
    marginLeft: `${sliderValueNumeric * 10}%`,
  }}
>
  <div className="status-slider__arrow-border">
    <div
      className="status-slider__arrow"
      style={{
        borderTopColor: GRADIENT_COLORS[sliderValueNumeric],
      }}
    ></div>
  </div>
</div>
```

``` css
.status-slider {
  &__arrow-wrapper {
    align-self: baseline;
  }

  &__arrow-border {
    width: 0;
    height: 0;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-top: 10px solid white;
    position: relative;
    /* presses the arrow into the slider */
    margin-bottom: -2px;
    /* places the point of the arrow at the very beginning of the slider by default so that you don't have to do extra math */
    margin-left: -10px;
  }

  &__arrow {
    width: 0;
    height: 0;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-top: 5px solid black;
    position: absolute;
    bottom: 3px;
    left: -5px;
  }
}
```

Reply:
Great idea, I appreciate your eye for details. I fixed it.